### PR TITLE
DOC: better installation instructions

### PR DESCRIPTION
**Pull request by matthew-brett**
Don't advise sudo, it's a terrible idea.

Make clear the difference between a "source" install and a "development"
install.

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -48,11 +48,16 @@ the source from `NiBabel pypi`_ .  Go to the pypi page and select the source
...
-or (if you need root permission to install on a unix system)::
+If you get permission errors, this may be because ``pip`` is trying to install
+to the system directories.  You can solve this error by using ``sudo``, but we
+strongly suggest you either do an install into your "user" directories, like
+this::
 
-    sudo python setup.py install
+    pip install --user .
+
+or you work inside a virtualenv_.
```
Comment (yarikoptic) [inline doc/source/installation.rst]: +1 fwiw ;-)

File: doc/source/installation.rst [hunk 2]
```diff
@@ -128,8 +124,10 @@ It should look something like this::
     >>>
 
 
-To run the nibabel test suite, from the terminal run ``nosetests nibabel`` or ``python -c "import nibabel; nibabel.test()``.
+To run the nibabel test suite, from the terminal run ``nosetests nibabel`` or
+``python -c "import nibabel; nibabel.test()``.
 
-To run an extended test suite that validates ``nibabel`` for long-running and resource-intensive cases, please see :ref:`advanced_testing`.
+To run an extended test suite that validates ``nibabel`` for long-running and
+resource-intensive cases, please see :ref:`advanced_testing`.
```
Comment (yarikoptic) [inline doc/source/installation.rst]: I didn't know... Need to read on ;-)

File: doc/source/installation.rst [hunk 3]
```diff
@@ -66,60 +71,51 @@ their repositories. Once this is done, installing NiBabel is::
...
-*  `Python Imaging Library`_ (for PNG conversion in DICOMFS)
-*  nose_ 0.11 or greater (to run the tests)
-*  sphinx_ (to build the documentation)
-
-Get the sources
----------------
+*  SciPy_ (optional, for full SPM-ANALYZE support)
+*  PyDICOM_ 0.9.7 or greater (optional, for DICOM support)
+*  `Python Imaging Library`_ (optional, for PNG conversion in DICOMFS)
+*  nose_ 0.11 or greater (optionaal, to run the tests)
```
Comment (yarikoptic) [inline doc/source/installation.rst]: Typo, optional

**Comment by matthew-brett** [comment]
I'm not sure what to make of the Donald Stufft article - I guess that we should expect `requirements.txt` to be used by developers, mainly.

The datalad idea as I understand it is to add `full` installation group, and then use the local development idea of the `requirements.txt` file to trigger that.

So, yes, maybe.  I wish I had a clearer feeling for how `requirements.txt` is used in practice.  But maybe Donald Stufft is right, that it is really only useful for people writing web apps with very strict requirements for deployment.

**Comment by yarikoptic** [comment]
On Sat, 06 Aug 2016, Matthew Brett wrote:

>    I'm not sure what to make of the Donald Stufft article - I guess that we
>    should expect requirements.txt to be used by developers, mainly.

or more "to pin point specific versions in particular deployments"

>    The datalad idea as I understand it is to add full installation group, and
>    then use the local development idea of the requirements.txt file to
>    trigger that.

yeap.  This way, for instance, pip install nibabel could install the
core of nibabel, and then additional  pieces could be pulled in   pip
install nibabel[dicom], or a full installation with everything
imaginable supported by nibabel  via 'nibabel[full]'

**Comment by matthew-brett** [comment]
OK to leave better requirements handling for a later PR?

**Comment by yarikoptic** [comment]
if you mean "let's merge this one now and improve later with another PR", then +1 on my side ;)

**Pull request merged**